src/chainstate/stacks/index/storage.rs

```diff
@@ -699,11 +765,40 @@ impl TrieFileStorage {
         Ok(())
     }
 
-    fn read_fork_table(dir_path: &String, root_hash: &BlockHeaderHash) -> Result<TrieForkTable, Error> {
-        // maps a block hash to its list of unique ancestors.
-        // has an entry for block hashes that are either chain tips, or parents of two or more forks.
-        let parent_children = TrieFileStorage::scan_blocks(dir_path)?;
-        TrieForkTable::new(root_hash, &parent_children)
+    fn read_block_hash_map(dir_path: &String, root_hash: &BlockHeaderHash) -> Result<(BlockHashMap, HashSet<BlockHeaderHash>), Error> {
+        let mut blocks = vec![];
+
+        let mut parents = HashSet::new();
+
+        TrieFileStorage::scan_blocks(dir_path, |block_name, block_path| {
+            let bhh = match BlockHeaderHash::from_hex(&block_name) {
```

Comment:
This is going to do interesting things to the non-optimized benchmarks.  We'll want to make sure that this method is super-fast when optimized (I think it's using `util::hash::from_hex` behind the scenes).

Reply:
What makes you say that? This is basically doing the exact same thing as the previous `scan_blocks` function (they both use `from_hex`) except that this function also reads the block identifier in addition to the parent. There is probably additional overhead in a non-optimized build for using an anonymous function, but the runtime costs of this function are probably dominated by the file opens in any case.

Reply:
Ah nm I commented in the wrong area.